### DISCUSSION TITLE
Bot responds 4 times

### DISCUSSION THREAD

**Discussion by neurlnet** [discordjs/discord.js#9564] [Q&A] [answered]
```js
client.on('messageCreate', (message) => {
    if(message.author.bot){
        return;
    }
    console.log("message received!")

   let joe = message.toString()
   joe = joe.toLowerCase();

   if(joe.startsWith(`what` , 0)|| joe.startsWith('which' ,  0) || joe.startsWith('how' , 0) || joe.startsWith('why', 0 ) || joe.startsWith('where' , 0)){
    const responses = ["I don't know!","Use /chat to get your answer", "I'm not sure!","Why not?","Because why not?","I don't care","Find it out on your own", "No","Nothing","Anything can be the answer!"]
   
    console.log(responses[Math.floor(Math.random() * 10)]);
    message.reply(responses[Math.floor(Math.random() * 10)]);
   }
})
```

So whenever the condition is satisfied, the bot replies 4 times
![image](https://github.com/discordjs/discord.js/assets/85750098/e1140537-ab99-4c99-80ca-32954690ac83)

How can I fix this?

**Comment by BenjammingKirby** [comment]
it seems like you either have multiple instances or nested event listeners, can you show how you have your messageCreate event listener

**Comment by neurlnet** [comment]
it is not nested, it is on the outer most level, this happens at times and then stops happening and it randomly occurs again

**Reply by BenjammingKirby** [reply]
i can assure you, code isn't magical, everything that happens does so for a reason, it must be because it's nested or is called twice or something, since the code is definitely ran multiple times

**Comment by Vaibhav-Naik-26** [comment, marked as answer]
```js
client.on('messageCreate', (message) => {
  if (message.author.bot) return;
  console.log("message received!");

  const joe = message.content.toLowerCase();

  if (joe.startsWith(`what`, 0) || joe.startsWith('which', 0) || joe.startsWith('how', 0) || joe.startsWith('why', 0) || joe.startsWith('where', 0)) {
    const responses = ["I don't know!", "Use /chat to get your answer", "I'm not sure!", "Why not?", "Because why not?", "I don't care", "Find it out on your own", "No", "Nothing", "Anything can be the answer!"]
   
    const response = responses[Math.floor(Math.random() * 10)]; // I put it in a different variable because you were calculating 2 times which was resulting in 2 different answers (BTW this is not the answer of your question, I just made the code simpler, easy and logical);
    console.log(response);
    message.reply(response);
   };
});
```
The above code is not the answer to your question but the code was a bit wrong.

Now the answer to your question is that the bot was connected 4 times at the same time (in short multiple instances), so the **normal solution** is to stop all the projects where the bot is logged in or the **ultimate solution** is Reset the Bot Token so that it will disconnect all the connections and now use the new token to login and it should reply the message only once.
And if after trying Ultimate Solution the problem exists then it means the Handler or some other code is wrong.

**Comment by neurlnet** [comment]
Thank you it works now!
